stellargraph/layer/graphsage.py

```diff
@@ -417,26 +457,34 @@ def _build_group_weights(self, in_shape, out_size, group_idx=0):
             weights = self.add_weight(
                 name=f"w_g{group_idx}",
                 shape=(int(in_shape[-1]), out_size),
-                initializer=self._initializer,
+                initializer=self.kernel_initializer,
+                regularizer=self.kernel_regularizer,
+                constraint=self.kernel_constraint,
                 trainable=True,
             )
         else:
             w_group = self.add_weight(
                 name=f"w_g{group_idx}",
                 shape=(self.hidden_dim, out_size),
-                initializer=self._initializer,
+                initializer=self.kernel_initializer,
+                regularizer=self.kernel_regularizer,
+                constraint=self.kernel_constraint,
                 trainable=True,
             )
             w_pool = self.add_weight(
                 name=f"w_pool_g{group_idx}",
                 shape=(int(in_shape[-1]), self.hidden_dim),
-                initializer=self._initializer,
+                initializer=self.kernel_initializer,
+                regularizer=self.kernel_regularizer,
+                constraint=self.kernel_constraint,
                 trainable=True,
             )
             b_pool = self.add_weight(
                 name=f"b_pool_g{group_idx}",
                 shape=(self.hidden_dim,),
-                initializer=self._initializer,
+                initializer=self.kernel_initializer,
```

Comment:
This is a bias for the pooling function. Applying kernel initializers, constraints and regularization here is interesting.  We really don't know the effects of regularization or constraints in these more complicated aggregation classes.

This is just a comment, I don't know what to do about it. Maybe add a comment in the code.

Reply:
In every case that I modified, there was a pre-existing use of `initializer=self._initializer` that I altered, so I am not in effect doing anything different, especially since the default case should have exactly the same behaviour. I take your point, however, that the intention might differ.

Reply:
If it is really a bias term, should it take the bias regulariser, etc.? As I mentioned, the previous code in fact assumed 'glorot_uniform' initialisation by default. I can just hard wire it to that, if you prefer.

Reply:
Actually good point, for bias we should really be initializing them to zeros. Perhaps you can use the `bias_initializer` here and set the default for that to be zero:
https://datascience.stackexchange.com/questions/17987/how-should-the-bias-be-initialized-and-regularized/18145#18145

Reply:
@adocherty,  now that I have looked more closely at the code, you are right - it is definitely a bias term. Unless you think otherwise, I am going to apply the bias regularisers to it, including the default of 'zeros', which will be a change from previous behaviour. Do we have any demos that use the pooling aggregators?

Reply:
Also, should the AttentionalAggregator of GraphSAGE allow the attn_kernel_regularizer?

Reply:
Hi @adocherty , could you please give this PR another look over? I think I have resolved all your requests - the outstanding issue is my question above, re allowing attn_kernel_regularizer in GraphSAGE for the AttentionalAggregator case.

Reply:
> Unless you think otherwise, I am going to apply the bias regularisers to it, including the default of 'zeros', which will be a change from previous behaviour.

I agree.

> Do we have any demos that use the pooling aggregators?

We have no demos that use the pooling aggregators at the moment, so these changes won't be a problem.

> Also, should the AttentionalAggregator of GraphSAGE allow the attn_kernel_regularizer?

I don't think that we need to expose this at the GraphSAGE class level at the moment.  The AttentionalAggregator is still pretty experimental.